rdmd.d

```diff
@@ -862,8 +862,12 @@ string innerEvalCode(string[] eval)
     import std.string : join, stripRight;
     // assumeSafeAppend just to avoid unnecessary reallocation
     string code = eval.join("\n").stripRight.assumeSafeAppend;
-    if (code.length > 0 && code[$ - 1] != ';')
-        code ~= ';';
+    if (code.length > 1 && code[$ - 2 .. $] == "*/")
```

Comment:
This is such a huge hack, I'm not sure I would approve this.

I think realistically it *only* fixes the issue submitter's use case.

Reply:
Well, passing comments to eval is something that probably wasn't even considered so I'm not sure what a "non-hack" approach would do here for such comments.

Reply:
The non-hack approach would be to try to compile the program both with and without the `;`, or integrate with the frontend to lex the string and figure out if `;` is needed from the grammatical context. Obviously both impractical.

Another direction would be to add a `-revert=` flag to DMD to allow stray `;` again, remove the darn `-d` from `rdmd` and replace it with the `-revert=`, and then remove it after a few versions once everyone has seen the deprecation message.

I don't think the grammar allows for any constructs that would terminate a possibly-empty statement, right?

Reply:
> I don't think the grammar allows for any constructs that would terminate a possibly-empty statement, right?

This is really stupid and works only with expression statements but:
```d
void terminator(){}               // added by rdmd
void terminator(T)(lazy T t){t;}  // added by rdmd

void fun(){}

void main()
{
	fun

	.terminator();  // added by rdmd

	fun;

	.terminator();  // added by rdmd
}
```